gradle-jdks/src/main/java/com/palantir/gradle/jdks/JdkManager.java

```diff
@@ -109,7 +109,7 @@ private Path findJavaHome(Path temporaryJdkPath) {
             return files.filter(file -> Files.isRegularFile(file)
                             // macos JDKs have a `bin/java` symlink to `Contents/Home/bin/java`
                             && !Files.isSymbolicLink(file)
-                            && file.endsWith(Paths.get("bin/java")))
+                            && file.endsWith(Paths.get("bin/" + SystemTools.java())))
```

Comment:
I suspect we may need to take the environment-specific separator into account:
```suggestion
                            && file.endsWith(Paths.get("bin" + FileSystems.getDefault().getSeparator() + SystemTools.java())))
```

Alternatively we could use 
```java
SystemTools.java().equals(file.getFileName().toString())
    && "bin".equals(file.getParent() /* null check? */ .getFileName().toString());
````

Reply:
Paths.get may handle normalization for us, I'm not sure off the top of my head -- safest/easiest may be `Paths.get("bin", SystemTools.java())`